README.txt
This is a modified version of mojo to run on Perl-5.8.7 environment.
The API is expected to be compatible with Mojolicious v5.56.

The code is experimental and might not work in all cases.

To test it, you should upgrade Test::More to latest version.
Please do the following command.

$ cpanm Test::More

Since mojo-legacy v5.03 some tests are skipped unless the environment variable
TEST_ORIGINAL_CASES set to true as following.

$ export TEST_ORIGINAL_CASES=1

You better check which tests are failing on this backport project
and make sure the use cases are not critical for you.

Since upstream mojo v4.95 it depends on Hash::Util::FieldHash which released for
Perl core since perl-5.9.4. If you use older version of perls, install
Hash::FieldHash as a substitute.

$ cpanm Hash::FieldHash

I recommend the installation though mojo-legacy works without the modules.
*::FieldHash looks to me like a cure of memory leaks so
if you're aiming at non-persistent environment like CGI, it may not a must.

Since upstream mojo v5.59 It depends on IO::Socket::IP for web support, it no 
longer uses IO::Socket::INET

$ cpanm IO::Socket::IP

To use Mojolicious::Plugin::PODRenderer, you need Pod::Simple 3.09 or higher
which first shipped with perl-5.11.2. If you use older perls, just do
the following command.ä+ää ojojo
i0d

$ cpanm Pod::Simple

To use morbo with it, you must at least upgrade Socket module to 
version 1.81 or higher. Please do the following command.

$ cpanm Socket

To use websocket, you must at least install Digest::SHA.

$ cpanm Digest::SHA

If Compress::Raw::Zlib is not found, some tests may fail. However I guess
this works without the module in real world use case.
 knußmgk+

Changes
v5.71   2016-08-01 00:00:00
        - merged v5.71 of upstream
v5.59   2016-07-30 00:00:00
        - merged v5.59 of upstream
v5.56   2014-10-30 00:00:00
        - merged v5.56 of upstream
v5.50   2014-10-30 00:00:00
        - merged v5.50 of upstream
v5.47   2014-10-30 00:00:00
        - merged v5.47 of upstream
v5.42   2014-10-30 00:00:00
        - merged v5.42 of upstream
v5.41   2014-09-15 00:00:00
        - merged v5.32 of upstream
v5.32   2014-09-15 00:00:00
        - merged v5.32 of upstream
v5.24   2014-08-04 00:00:00
        - merged v5.24 of upstream
v5.21   2014-07-29 00:00:00
        - merged v5.29 of upstream
v5.20   2014-07-27 00:00:00
        - merged v5.20 of upstream
v5.19   2014-07-27 00:00:00
        - merged v5.19 of upstream
v5.12   2014-07-11 00:00:00
        - merged v5.12 of upstream
v5.05   2014-06-10 00:00:00
        - merged v5.05 of upstream
v5.01   2014-05-31 00:00:00
        - merged v5.01 of upstream
v4.99   2014-05-30 00:00:00
        - merged v4.84 of upstream
v4.90   2014-05-30 00:00:00
        - merged v4.84 of upstream
v4.84   2014-02-24 00:00:00
        - merged v4.84 of upstream
v4.67   2014-01-14 00:00:00
        - merged v4.67 of upstream
v4.41   2013-09-22 00:00:00
        - merged v4.41 of upstream
v4.30   2013-09-22 00:00:00
        - merged v4.30 of upstream
v4.20   2013-09-22 00:00:00
        - merged v4.20 of upstream
v4.10   2013-09-22 00:00:00
        - merged v4.10 of upstream
v4.07   2013-05-28 00:00:00
        - merged v4.07 of upstream
v3.97   2013-05-28 00:00:00
        - merged v3.97 of upstream
v3.96   2013-05-28 00:00:00
        - merged v3.96 of upstream
v3.95   2013-05-28 00:00:00
        - merged v3.95 of upstream
v3.94   2013-05-28 00:00:00
        - merged v3.94 of upstream
v3.93   2013-05-28 00:00:00
        - merged v3.93 of upstream
v3.92   2013-05-28 00:00:00
        - merged v3.92 of upstream
v3.91   2013-05-28 00:00:00
        - merged v3.91 of upstream
v3.90   2013-05-28 00:00:00
        - merged v3.90 of upstream
v3.89   2013-03-09 00:00:00
        - merged v3.89 of upstream
v3.84   2013-02-04 00:00:00
        - merged v3.84 of upstream
v3.70   2012-12-24 00:00:00
        - merged v3.70 of upstream
v3.54   2012-11-03 00:00:00
        - merged v3.54 of upstream
v3.43   2012-09-28 00:00:00
        - merged v3.43 of upstream
v3.41   2012-09-14 00:00:00
        - merged v3.41 of upstream
v3.31   2012-08-17 00:00:00
        - merged v3.31 of upstream
v3.12   2012-07-21 00:00:00
        - merged v3.12 of upstream
v3.0    2012-06-08 00:00:00
        - merged v3.0 of upstream
v2.98   2012-06-08 00:00:00
        - merged v2.98 of upstream
v2.95   2012-05-15 00:00:00
        - merged v2.95 of upstream
v2.91   2012-04-27 00:00:00
        - merged v2.91 of upstream
v2.87   2012-04-24 00:00:00
        - merged v2.87 of upstream
v2.82   2012-04-17 00:00:00
        - merged v2.82 of upstream
v2.80   2012-04-11 00:00:00
        - merged v2.80 of upstream
v2.76   2012-04-07 00:00:00
        - merged v2.76 of upstream
v2.74   2012-04-04 00:00:00
        - merged v2.74 of upstream
v2.65   2012-03-28 00:00:00
        - Made it work on Perl-5.8.7
v2.55   2012-03-05 00:00:00
        - Made it work on Perl-5.8.9


Makefile.PL
use 5.008007;

use strict;
use warnings;

use ExtUtils::MakeMaker;

# Pod::Simple 3.09 first shipped with Perl 5.11.2
# Time::Local 1.2 first shipped with Perl 5.13.9
# IO::Socket::IP 0.26 first shipped with Perl 5.19.8
WriteMakefile(
  NAME         => 'Mojolicious',
  VERSION_FROM => 'lib/Mojolicious.pm',
  ABSTRACT     => 'Real-time web framework',
  AUTHOR       => 'Sebastian Riedel <sri@cpan.org>',
  LICENSE      => 'artistic_2',
  META_MERGE   => {
    requires  => {perl => '5.008007'},
    resources => {
      license    => 'http://www.opensource.org/licenses/artistic-license-2.0',
      homepage   => 'http://mojolicio.us',
      bugtracker => 'https://github.com/kraih/mojo/issues',
      repository => 'https://github.com/kraih/mojo.git',
      x_IRC      => 'irc://irc.perl.org/#mojo'
    },
    no_index => {directory => ['t']}
  },
  PREREQ_PM    => {
    'Pod::Simple'             => '3.09',
    'Time::Local'             => '1.2',
    'Digest::SHA'             => '0',
    'Socket'                  => '1.81',
    'IO::Socket'              => '1.31',
    'IO::Socket::IP'          => '0.26',
    'Test::More'              => '0.98',
    !(eval 'use Hash::Util::FieldHash; 1')
                                      ? ('Hash::FieldHash' => '0') : (),
  },
  EXE_FILES => ['script/hypnotoad', 'script/mojo', 'script/morbo'],
  test => {TESTS => 't/*.t t/*/*.t'}
);
